pkg/astrolabe/protected_entity.go

```diff
@@ -60,16 +60,16 @@ func NewProtectedEntityIDFromModel(mpei models.ProtectedEntityID) (ProtectedEnti
 	return NewProtectedEntityIDFromString(string(mpei))
 }
 
-func NewProtectedEntityIDWithNamespace(peType string, name string, namespace string) ProtectedEntityID {
+func NewProtectedEntityIDWithNamespaceAndSnapshot(peType string, name string, namespace string, snapshotID string) ProtectedEntityID {
```

Comment:
Rather than force the callers that don't have snapshots to call this with "", why don't you leave the original NewProtectedEntityIDWithNamespace and have it call NewProtectedEntityIDWithNamespaceAndSnapshot with a blank snapshot ID?